File: Data Structures & Algorithms/invert-a-binary-tree/submission-3.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        if not root:
            return None
        def swap(node):
            left = node.left
            node.left = node.right
            node.right = left
            if node.left:
                swap(node.left)
            if node.right:
                swap(node.right)
            return node
        node = swap(root)
        return node

File: Data Structures & Algorithms/binary-tree-maximum-path-sum/submission-3.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    int max;
    private int compute(TreeNode node) {
        if (node == null) {
            return 0;
        }
        int left = compute(node.left);
        int right = compute(node.right);
        int curr = Math.max( node.val + Math.max(left, right), Math.max(node.val, node.val + left + right));
        max = Math.max(curr, max);
        return Math.max(node.val, node.val + Math.max(left, right));
    }
    public int maxPathSum(TreeNode root) {
        max = -1001;
        compute(root);
        return max;
        
    }
}
